cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -269,14 +269,16 @@ private int processOneChangesBatch(ChangesResultWrapper changeFeeds)
 
             if (this.cancel) { break; }
 
-            List<Callable<DocumentRevsList>> tasks = createTasks(batch, missingRevisions);
+            Callable<Iterable<DocumentRevsList>> tasks = createTask(batch, missingRevisions);
             try {
-                List<Future<DocumentRevsList>> futures = executor.invokeAll(tasks);
-                for(Future<DocumentRevsList> future : futures) {
-                    DocumentRevsList result = future.get();
+                Future<Iterable<DocumentRevsList>> future = executor.submit(tasks);
 
+                Iterable<DocumentRevsList> result = future.get();
```

Comment:
`future.get()` is a blocking call and this now appears to be a single task submitted for execution on the previous line, so if there is no parallel processing to be gained what is the value of using the `executor` here?